### PR TITLE
[15.0][MIG] analytic_base_department: Migration to 15.0

### PR DESCRIPTION
**Pull request by Camille0907**
Migration of the module from version 14 to version 15.

### CODE REVIEW

File: analytic_base_department/models/analytic.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class AnalyticAccount(models.Model):
+    """Add Department in analytic account."""
+
+    _inherit = "account.analytic.account"
+
+    department_id = fields.Many2one(comodel_name="hr.department")
```
Comment (BT-anieto) [inline analytic_base_department/models/analytic.py]: I would add a String for the translations

File: analytic_base_department/models/analytic.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    @api.model
+    def _default_department(self):
+        department_id = False
+        employee = self.env.user.employee_ids
+        if employee and employee[0].department_id:
+            department_id = employee[0].department_id.id
+        return department_id
+
+    department_id = fields.Many2one(
```
Comment (BT-anieto) [inline analytic_base_department/models/analytic.py:28]: Same here, I would add a string

File: analytic_base_department/models/hr_department.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2020 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import fields, models
+
+
+class Department(models.Model):
+    _inherit = "hr.department"
+
+    analytic_account_ids = fields.One2many(
```
Comment (BT-anieto) [inline analytic_base_department/models/hr_department.py:10]: Same here, a string.

**Review by BT-anieto — APPROVED** [review]
I would add the 'string' in the definition of a field to have the label translated everywhere, but not critical.

File: analytic_base_department/tests/test_line_department.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+from odoo.tests.common import TransactionCase
+
+
+class LineDepartmentCase(TransactionCase):
+    def setUp(self):
+        """Assign user and department."""
+        super(LineDepartmentCase, self).setUp()
+        # base.user_demo --> hr.employee_qdp --> hr.dep_rd
+        self.user = self.env.ref("base.user_demo")
+        self.dep = self.env.ref("hr.dep_rd")
```
Comment (BT-cserra) [inline analytic_base_department/tests/test_line_department.py]: This should be now [`setUpClass`](https://github.com/odoo/odoo/blob/d4e7eb99818cd6780c3e0a94a9045951ac3708a7/odoo/tests/common.py#L698).

```suggestion
    @classmethod
    def setUpClass(cls):
        """Assign user and department."""
        super(LineDepartmentCase, cls).setUpClass()
        # base.user_demo --> hr.employee_qdp --> hr.dep_rd
        cls.user = cls.env.ref("base.user_demo")
        cls.dep = cls.env.ref("hr.dep_rd")
```

**Review by BT-cserra — APPROVED** [review]

File: analytic_base_department/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2011-2018 Camptocamp SA
+# Copyright 2013 Daniel Reis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Base Analytic Department Categorization",
+    "summary": "Add relationshet between Analytic and Department",
```
Comment (florentx) [inline analytic_base_department/__manifest__.py]: Please fix typo:
```suggestion
    "summary": "Add relationship between Analytic and Department",
```

File: analytic_base_department/views/analytic.xml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                <field name="department_id" />
+            </field>
+        </field>
+    </record>
+    <record id="view_account_analytic_account_search" model="ir.ui.view">
+        <field name="name">account.analytic.account.search</field>
+        <field name="model">account.analytic.account</field>
+        <field name="inherit_id" ref="analytic.view_account_analytic_account_search" />
+        <field name="arch" type="xml">
+            <xpath expr="//search/group" position="inside">
```
Comment (florentx) [inline analytic_base_department/views/analytic.xml:29]: Please forward port this PR which was merged in 13.0 (but not ported to 14.0).
OCA/account-analytic#364

**Comment by BT-anieto** [comment]
Hi! :)  Could you do the small fix so that it can be merged? Thx in advance.

**Comment by Camille0907** [comment]
> Hi! :) Could you do the small fix so that it can be merged? Thx in advance.

Done :)

**Review by sebalix — APPROVED** [review]

**Pull request merged**